core/parser.mly

```diff
@@ -839,26 +839,27 @@ mutual_bindings:
 | binding                                                      { MutualBindings.(add (empty (pos $loc)) $1) }
 | mutual_bindings binding                                      { MutualBindings.add $1 $2 }
 
-bindings:
+binding_or_mutual:
 | binding                                                      { [$1]      }
 | mutual_binding_block                                         { $1        }
-| bindings mutual_binding_block                                { $1 @ $2   }
-| bindings binding                                             { $1 @ [$2] }
+
+bindings:
+| binding_or_mutual                                            { $1 } /* See #441 and #900 */
+| bindings binding_or_mutual                                   { $1 @ $2 }
 
 moduleblock:
 | LBRACE declarations RBRACE                                   { $2 }
 
 block:
 | LBRACE block_contents RBRACE                                 { block ~ppos:$loc $2 }
 
-block_contents:
-| bindings exp SEMICOLON                                       { ($1 @ [with_pos $loc($2) (Exp $2)],
-                                                                  record ~ppos:$loc []) }
+case_contents:
 | bindings exp                                                 { ($1, $2) }
-| exp SEMICOLON                                                { ([with_pos $loc($1) (Exp $1)],
-                                                                  record ~ppos:$loc []) }
 | exp                                                          { ([], $1) }
-| SEMICOLON | /* empty */                                      { ([], with_pos $loc (TupleLit [])) }
+
+block_contents:
+case_contents                                                  { $1 }
```

Comment:
We always start the first production with a leading `| `.

Reply:
Done